File: core-io/src/main/java/com/couchbase/client/core/env/RoleBasedCredentials.java
package com.couchbase.client.core.env;

public class RoleBasedCredentials implements Credentials {

  private final String username;
  private final String password;

  public RoleBasedCredentials(String username, String password) {
    this.username = username;
    this.password = password;
  }

  @Override
  public String usernameForBucket(String bucket) {
    return username;
  }

  @Override
  public String passwordForBucket(String bucket) {
    return password;
  }

}

File: scala-client/src/main/scala/com/couchbase/client/scala/env/RoleBasedCredentials.scala
package com.couchbase.client.scala.env

import com.couchbase.client.core.env.Credentials

case class RoleBasedCredentials(username: String, password: String) extends Credentials {
  override def usernameForBucket(bucket: String): String = username
  override def passwordForBucket(bucket: String): String = password
}
